pkg/apis/batch/v1/defaults_test.go

```diff
@@ -115,47 +123,95 @@ func TestSetDefaultJob(t *testing.T) {
 			},
 			expected: &batchv1.Job{
 				Spec: batchv1.JobSpec{
-					Completions: newInt32(2),
-					Parallelism: newInt32(1),
+					Completions:     newInt32(2),
+					Parallelism:     newInt32(1),
+					BackoffLimit:    newInt32(1),
+					FailedPodsLimit: newInt32(1),
+				},
+			},
+			expectLabels: true,
+		},
+		"BackoffLimit explicitly 2 and others unset -> parallelism is defaulted": {
+			original: &batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: newInt32(2),
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{Labels: defaultLabels},
+					},
+				},
+			},
+			expected: &batchv1.Job{
+				Spec: batchv1.JobSpec{
+					Completions:     newInt32(1),
+					Parallelism:     newInt32(1),
+					BackoffLimit:    newInt32(2),
+					FailedPodsLimit: newInt32(1),
+				},
+			},
+			expectLabels: true,
+		},
+		"FailedPodsLimit explicitly 2 and others unset -> parallelism is defaulted": {
+			original: &batchv1.Job{
+				Spec: batchv1.JobSpec{
+					FailedPodsLimit: newInt32(2),
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{Labels: defaultLabels},
+					},
+				},
+			},
+			expected: &batchv1.Job{
+				Spec: batchv1.JobSpec{
+					Completions:     newInt32(1),
+					Parallelism:     newInt32(1),
+					BackoffLimit:    newInt32(2),
+					FailedPodsLimit: newInt32(1),
```

Comment:
Right, 
The fact is that I mess up some commits split. I did include in this commit the test logic update that is checking the value: BackoffLimit and FailedPodsLimit.